common/ipfs/api.go

```diff
@@ -0,0 +1,99 @@
+package ipfs
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/abevier/tsk/ratelimiter"
+
+	iface "github.com/ipfs/boxo/coreiface"
+	"github.com/ipfs/kubo/client/rpc"
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/ceramicnetwork/go-cas/models"
+)
+
+const defaultIpfsRateLimit = 16
+const defaultIpfsBurstLimit = 16
+const defaultIpfsLimiterMaxQueueDepth = 100
+const defaultIpfsPublishPubsubTimeoutS = 30
+
+type PubSubPublishTask struct {
+	Topic string
+	Data  []byte
+}
+
+type IpfsApi struct {
+	core            iface.CoreAPI
```

Comment:
I've been thinking about this the last few days. Since we only ever want any core API calls to go through the rate limiter, and never otherwise, we should probably hide it from outside callers. We'll have to expose its APIs one-by-one as we need them but that should be ok because we likely won't need to expose that many anyway.